src/Data/Language.hs

```diff
@@ -111,53 +20,15 @@ supportedExts = foldr append mempty supportedLanguages
     supportedLanguages = fmap lookup (languageToText <$> codeNavLanguages)
     lookup k = Map.lookup k Lingo.languages
 
-codeNavLanguages :: [Language]
-codeNavLanguages = [Go, Java, Ruby, Python, JavaScript, TypeScript, PHP]
-
-pathIsMinified :: FilePath -> Bool
-pathIsMinified = isExtensionOf ".min.js"
-
-languageToText :: Language -> T.Text
-languageToText = \case
-  Unknown -> "Unknown"
-  Go -> "Go"
-  Haskell -> "Haskell"
-  Java -> "Java"
-  JavaScript -> "JavaScript"
-  JSON -> "JSON"
-  JSX -> "JSX"
-  Markdown -> "Markdown"
-  Python -> "Python"
-  Ruby -> "Ruby"
-  TypeScript -> "TypeScript"
-  TSX -> "TSX"
-  PHP -> "PHP"
-
-textToLanguage :: T.Text -> Language
-textToLanguage = \case
-  "Go" -> Go
-  "Haskell" -> Haskell
-  "Java" -> Java
-  "JavaScript" -> JavaScript
-  "JSON" -> JSON
-  "JSX" -> JSX
-  "Markdown" -> Markdown
-  "Python" -> Python
-  "Ruby" -> Ruby
-  "TypeScript" -> TypeScript
-  "TSX" -> TSX
-  "PHP" -> PHP
-  _ -> Unknown
-
 
 data PerLanguageModes = PerLanguageModes
-  { pythonMode :: LanguageMode
-  , rubyMode   :: LanguageMode
-  , goMode     :: LanguageMode
+  { pythonMode     :: LanguageMode
+  , rubyMode       :: LanguageMode
+  , goMode         :: LanguageMode
   , typescriptMode :: LanguageMode
-  , tsxMode :: LanguageMode
+  , tsxMode        :: LanguageMode
   , javascriptMode :: LanguageMode
-  , jsxMode :: LanguageMode
+  , jsxMode        :: LanguageMode
```

Comment:
The `LanguageMode` stuff still lives inside `Data.Language` because it’s a `semantic`-specific concern.